### PR TITLE
KNL-1502 Cache raw type, memoryService.newCache()

### PR DESCRIPTION
**Pull request by axxter99**
https://jira.sakaiproject.org/browse/KNL-1502

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/DbAuthzGroupService.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
 	/** Our log (commons). */
 	private static Logger M_log = LoggerFactory.getLogger(DbAuthzGroupService.class);
 	/** All the event functions we know exist on the db. */
-	protected Collection m_functionCache = new HashSet();
+	protected Collection<Object> m_functionCache = new HashSet<>();
```
Comment (buckett) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/DbAuthzGroupService.java]: I think `m_functionCache` can be limited to just `String` as it should only contain function names.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BasePreferencesService.java [hunk 2]
```diff
@@ -803,7 +803,7 @@ public BasePreferences(String id)
 			ResourcePropertiesEdit props = new BaseResourcePropertiesEdit();
 			m_properties = props;
 
-			m_props = new Hashtable();
+			m_props = new Hashtable<String, ResourcePropertiesEdit>();
```
Comment (buckett) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BasePreferencesService.java]: I think you can just use the `<>` operator here.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BasePreferencesService.java [hunk 3]
```diff
@@ -833,7 +833,7 @@ public BasePreferences(Element el)
 			// setup for properties
 			m_properties = new BaseResourcePropertiesEdit();
 
-			m_props = new Hashtable();
+			m_props = new Hashtable<String, ResourcePropertiesEdit>();
```
Comment (buckett) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BasePreferencesService.java]: I think you can just use the `<>` operator here.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BasePreferencesService.java [hunk 4]
```diff
@@ -938,14 +938,14 @@ protected void setAll(Preferences prefs)
 			m_properties.addAll(prefs.getProperties());
 
 			// %%% is this deep enough? -ggolden
-			m_props = new Hashtable();
+			m_props = new Hashtable<String, ResourcePropertiesEdit>();
```
Comment (buckett) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BasePreferencesService.java]: I think you can just use the `<>` operator here.

**Comment by axxter99** [comment]
@buckett cheers!

**Pull request merged**